pennylane/transforms/core/transform.py

```diff
@@ -26,40 +27,77 @@ def transform(
 ):
-    """The transform function is to be used to validate and dispatch a quantum transform on PennyLane objects (tape,
-    qfunc and Qnode). It can be used directly as a decorator on qfunc and qnodes.
+    """The transform function is to be used to validate and dispatch a quantum transform on PennyLane objects
+    (:class:`pennylane.tape.QuantumTape`, quantum function and :class:`pennylane.QNode`). After applying the function on
+    a valid quantum transform, the resulted function can be used on all the mentioned objects. The function can be used
+    directly as a decorator on quantum function and :class:`pennylane.QNode`.
 
     Args:
         quantum_transform (callable): A quantum transform is defined as a function that has the following requirements:
 
-            * A quantum transform is a function that takes a quantum tape as first input and returns a sequence of tapes
-              and a processing function.
+            * A quantum transform is a function that takes a :class:`pennylane.tape.QuantumTape` as first input and
+              returns a sequence of :class:`pennylane.tape.QuantumTape` and a processing function.
 
-            * The transform must have type hinting of the following form: my_quantum_transform(tape:
-              qml.tape.QuantumTape, ...) -> ( Sequence[qml.tape.QuantumTape], callable)
+            * The transform must have type hinting of the following form: ``my_quantum_transform(tape:
+              qml.tape.QuantumTape, ...) -> ( Sequence[qml.tape.QuantumTape], callable)``
 
         expand_transform (callable): An expand transform is defined as a function that has the following requirements:
 
             * An expand transform is a function that is applied before applying the defined quantum transform. It
-              takes the same arguments as the transform and returns a single tape in a sequence with a dummy processing
-              function.
+              takes the same arguments as the transform.
 
             * The expand transform must have the same type hinting as a quantum transform.
 
-        classical_cotransform (callable): A classical co-transform. NOT YET SUPPORTED.
+        classical_cotransform (callable): A classical co-transform is a function to post-process the the classical
+            jacobian and the quantum jacobian and has the signature: ``my_cotransform(qjac, cjac, tape) -> tensor_like``
         is_informative (bool): Whether or not a transform is informative. If true the transform is queued at the end
             of the transform program and the tapes or qnode aren't executed.
         final_transform (bool): Whether or not the transform is terminal. If true the transform is queued at the end
             of the transform program. ``is_informative`` supersedes ``final_transform``.
 
+    Returns:
+
+        pennylane.transforms.core.TransformDispatcher: It returns a transform dispatcher object that is dispatching
+            given the received circuit object.
```

Comment:
formatting was messed up because of newline
```suggestion
        pennylane.transforms.core.TransformDispatcher: It returns a transform dispatcher object that is dispatchin given the received circuit object.
```